### DISCUSSION TITLE
Using SwapChainPanel in C++ WinUI3 app

### DISCUSSION THREAD

**Discussion by robhurry** [microsoft/WindowsAppSDK#865] [Q&A] [answered]
I'm trying to create an application using the XAML SwapChainPanel in WinUI3 and Project Reunion to draw from directly to the panel using DirectX.
The sample applications are all based on UWP applications and I have translated them as best I can (Such as using the DispatcherQueue in place of the CoreDispatcher). 
My proble is regardless of what I've tried, when I attempt to get the Native Swap Chain instance of the panel using the following code:
```
winrt::Microsoft::System::DispatcherQueue dispatcher = swPanel().DispatcherQueue();
    dispatcher.TryEnqueue(winrt::Microsoft::System::DispatcherQueuePriority::High, winrt::Microsoft::System::DispatcherQueueHandler([=]() {

        ISwapChainPanelNative* panelNative;
        reinterpret_cast<IUnknown*>(&swPanel())->QueryInterface(__uuidof(ISwapChainPanelNative), (void**)&panelNative);
        }));
```
 I get the following error:
'SwapChainTest.exe' (Win32): Loaded 'C:\Windows\SysWOW64\avrt.dll'. 
'SwapChainTest.exe' (Win32): Loaded 'C:\Program Files\WindowsApps\Microsoft.ProjectReunion.0.5_0.52105.10000.0_x86__8wekyb3d8bbwe\Microsoft.DirectManipulation.dll'. 
'SwapChainTest.exe' (Win32): Loaded 'C:\Windows\SysWOW64\d2d1debug3.dll'. 
'SwapChainTest.exe' (Win32): Loaded 'C:\Windows\SysWOW64\DXGIDebug.dll'. 
'SwapChainTest.exe' (Win32): Loaded 'C:\Windows\SysWOW64\d3d11_3SDKLayers.dll'. 
Exception thrown at 0x042EB960 (Microsoft.ui.xaml.dll) in SwapChainTest.exe: 0xC0000005: Access violation executing location 0x042EB960

I've kept things extremely simply following the samples as best I can but can't seem to get the QueryInstance to work.

Thanks!

**Comment by robhurry** [comment, marked as answer]
Well for it's worth the answer was to use this syntax rather. 

` ISwapChainPanelNative* panelNative;

  winrt::get_unknown(testPanel())->QueryInterface(__uuidof(ISwapChainPanelNative), (void**)&panelNative);`

As an aside having samples built using WinUI 3 would be really great!

**Comment by ifgotmoney** [comment]
Can we just have more example of using directX & SwapChainPanel in C++/Winrt and WinUI3 for Desktop Apps?
